### DISCUSSION TITLE
Stellarium Script

### DISCUSSION THREAD

**Discussion by hagastro** [Stellarium/stellarium#4009] [General]
How do you change the "DSO filter by type settings" from a script

**Comment by alex-w** [comment]
It's a bit difficult, but see the docs for `NebulaMgr::setTypeFilters()` function

**Reply by hagastro** [reply]
Thanks, one more question. I have searched the script pages but cannot find a way to change the config.ini flag "flag_nebula_display_no_texture" from a script . Can you point me in the right direction please.

**Reply by alex-w** [reply]
Obvious answer - https://stellarium.org/doc/24.0/classStelSkyLayerMgr.html#a03d8c956d444a34c13a0475dc48e4672

`StelSkyLayerMgr.setFlagShow(false);`

**Comment by hagastro** [comment]
Many thanks for your recent help. I have used Stellarium for many years and demonstrate and promote it at the many public talks I give.
I have been asked to write a self running script for a tour of the night sky, with or without user control. I have solved the problem of pausing and starting the script from the keyboard but I was asked if an audio commentary could be added. It seems that StelAudioMgr is now disabled. Is there any way to do this. It would be very useful for presentations.

Many thanks,
Steve Broadbent BSc FRAS
Hampshire Astronomical Group
Clanfield Observatory
Mobile: 07813 086215
https://hantsastro.org.uk<https://hantsastro.org.uk/>

**Reply by gzotti** [reply]
Just tested 24.4/Win10. No problems with audio, plays WAV/OGG/MP3.

**Comment by hagastro** [comment]
Running 24.4 under Windows 11 Pro 24H2
Script:
StelAudioMgr.loadSound("audio/Recording.mp3","test");
StelAudioMgr.playSound("test");
Log:
Starting script at Thu Jan 9 13:02:26 2025
script error: 'Reference' @ line 2: ReferenceError: StelAudioMgr is not defined
Script finished at Thu Jan 9 13:02:26 2025

SteveB
Mobile: 07813 086215

**Reply by gzotti** [reply]
Ah, no. StelAudioMgr is not scriptable. 

core.loadSound("tests/audiotest.mp3", "mp3");
core.playSound("mp3");
core.wait(4);
core.dropSound("mp3");

**Comment by hagastro** [comment]
Many thanks.

SteveB
Mobile: 07813 086215
